common/djangoapps/philu_commands/management/commands/autoscore_ondemand_course_ora.py

```diff
@@ -64,41 +64,46 @@ def handle(self, *args, **options):
-
-                                                # check if this chapter is 2 weeks older or not.
-                                                module_access_days = delta_days.days - (index_chapter * 7)
-
-                                                if module_access_days >= DAYS_TO_WAIT_AUTO_ASSESSMENT \
-                                                        and response_submission_delta.days >= DAYS_TO_WAIT_AUTO_ASSESSMENT:
-                                                    try:
-                                                        # Status[0] is the status of assessment that are in waiting mode
-                                                        AssessmentWorkflow.objects.get(
-                                                            status=AssessmentWorkflow.STATUSES[0],
-                                                            course_id=course.id, item_id=block,
-                                                            submission_uuid=response_submissions.uuid)
-                                                        student = {'id': user.id, 'username': user.username,
-                                                                   'email': user.email,
-                                                                   'anonymous_user_id': anonymous_user.anonymous_user_id
-                                                                   }
-                                                        autoscore_ora(course.id, unicode(block), student)
-                                                    except AssessmentWorkflow.DoesNotExist:
-                                                        continue
-
-                                            except Submission.DoesNotExist:
+                                            response_submission = Submission.objects.filter(
+                                                student_item__student_id=anonymous_user.anonymous_user_id,
+                                                student_item__item_id=block).first()
+                                            if not response_submission:
                                                 continue
+                                            log.info('Response Created at: {created_date}'.format(
+                                                created_date=response_submission.created_at.date()
+                                            ))
+
+                                            response_submission_delta = today - response_submission.created_at.date()
+
+                                            # check if this chapter is 2 weeks older or not.
+                                            module_access_days = delta_days.days - (index_chapter * 7)
+
+                                            if (module_access_days >= DAYS_TO_WAIT_AUTO_ASSESSMENT and
+                                                    response_submission_delta.days >= DAYS_TO_WAIT_AUTO_ASSESSMENT):
```

Comment:
In all these lines (`75 to 90`) we are checking if this particular assessment qualifies to auto score or not, How about moving this logic in a separate method? I know this was already done the same way and your change is just to use filter instead of get to fetch submissions, but this small effort can make this command more readable.